multicluster/controllers/multicluster/serviceexport_controller.go

```diff
@@ -562,9 +571,9 @@ func (r *ServiceExportReconciler) updateOrCreateResourceExport(resName string,
 	ctx context.Context,
 	req ctrl.Request,
 	newResExport *mcsv1alpha1.ResourceExport,
-	existResExport *mcsv1alpha1.ResourceExport,
+	existingResExport *mcsv1alpha1.ResourceExport,
 	rc commonarea.RemoteCommonArea) error {
-	createResExport := reflect.DeepEqual(*existResExport, mcsv1alpha1.ResourceExport{})
+	createResExport := reflect.DeepEqual(*existingResExport, mcsv1alpha1.ResourceExport{})
 	resNamespaced := types.NamespacedName{Namespace: rc.GetNamespace(), Name: resName}
 	if createResExport {
 		newResExport.Finalizers = []string{common.ResourceExportFinalizer}
```

Comment:
Because there is another resource 'ResourceImport' in the leader cluster need to be clean up or updated  when there is a ResourceExport deletion event. if we delete ResourceExport directly when the ServiceExport is deleted, then leader reconciler will not be able to get the details of deleted ResourceExport, which we need to do ResourceImport deletion or udpate. Here is an intro of Finalizer in kubebuilder https://book.kubebuilder.io/reference/using-finalizers.html

Reply:
Ok. Better to add some comments about this.

Reply:
sure, I added a few information in ResoureExport controller, let me add it here as well.

Reply:
done